libtiledbsc/include/tiledbsc/soma.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SOMA_H
+#define SOMA_H
+
+#include <tiledb/tiledb>
+#include <tiledb/tiledb_experimental>
+
+namespace tiledbsc {
+using namespace tiledb;
+
+class SOMA {
+   public:
+    //===================================================================
+    //= public static
+    //===================================================================
+    static SOMA open(std::string_view uri);
+
+    //===================================================================
+    //= public non-static
+    //===================================================================
+    SOMA(std::string_view uri);
+
+    std::map<std::string, std::string> list_arrays();
+
+   private:
+    //===================================================================
+    //= private static
+    //===================================================================
+
+    //===================================================================
+    //= private non-static
+    //===================================================================
+    Context ctx_;
+    std::string uri_;
+    bool group_uri_override_ = false;
```

Comment:
Can we add a quick note (for those reading only the `.h` on a flyover) what this is for?